.github/workflows/pipeline-release.yml

```diff
@@ -9,38 +9,66 @@ on:
         default: '0.0.0'
 
 jobs:
+  prepare-release:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: create release branch
+      run: |
+        git checkout -b release/${{ github.event.inputs.version }}
+
+    - name: assign cmake version
+      run: "((Get-Content -Path CMakeLists.txt -Raw) -Replace 'VERSION 0.1.0 # local build version','VERSION ${{ github.event.inputs.version }}') | Set-Content -Path CMakeLists.txt"
```

Comment:
Is there a more elegant way to do this? I guess `Set-Content` will also change the encoding of `CMakeLists.txt`.

Reply:
There is some tool in GitHub Marketplace to replace strings, but they are not GitHub 1P repos. Any good idea?
Adding `NoNewLine` switch can stop `Set-Content` from appending an extra line.